orders-core/src/test/java/io/helidon/examples/sockshop/orders/CDIMocks.java

```diff
@@ -0,0 +1,82 @@
+package io.helidon.examples.sockshop.orders;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Member;
+import java.lang.reflect.Type;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.Annotated;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
+import javax.enterprise.inject.spi.ProcessInjectionPoint;
+import javax.enterprise.inject.spi.WithAnnotations;
+
+/**
+ * @RegisterRestClient annotation is processed by Jersey to create a Bean that implements
+ * the given interface. Adding test instances that mock interaction with the service causes
+ * ambiguity. This Extension finds InjectionPoints and modifies the set of Qualifiers, so
+ * the one annotated with @TestInstance is selected by CDI.
+ */
+public class CDIMocks implements Extension {
+   protected Set annotated = new HashSet();
+
+   public void collectCDIMocks(@Observes
+                               @WithAnnotations({Mock.class})
+                               ProcessAnnotatedType<?> pat) {
+       Class<?> clazz = pat.getAnnotatedType().getJavaClass();
+       if (clazz.getInterfaces().length == 0) {
+          System.out.println("FOUND ANNOTATED CLASS, BUT IT DOES NOT IMPLEMENT ANYTHING - WILL NOT MOCK " + clazz);
```

Comment:
I don't agree about the FINEST level. It is done once at test startup, and it is better to see what annotations kick in, and what gets skipped, so you don't need to dig too deep to find out why some test is failing.

Reply:
By all means, make it INFO level then. It only affects tests anyway, but it still should be replaced with proper logging instead of `System.out` calls.